v_0_1_0/master_template.go

```diff
@@ -833,7 +833,8 @@ write_files:
       strategy:
         type: RollingUpdate
         rollingUpdate:
-          maxUnavailable: 1
+          maxSurge: 1
+          maxUnavailable: 0
```

Comment:
what means maxUnavaiable 0? does that work?
we have nginx controller limited to 1 per machine  and there can be created another pod of we dont remove the previous

Reply:
@calvix this comes too from the recommended settings in the linked issues, I've checked it and seems to work, deleted a ic pod, it moves to terminating and a new one is created while the previous one is still there, eventually the old one is gone and then the new one moves from pending to running, is that ok?

Reply:
as we don't have hard-anti-affitini it should work

Reply:
yep, also these settings make more sense imo than maxUnavailable=1 (given that only one runs on each host), they try make sure that there will be always (at most) one pending while any of them is terminating.

Reply:
ack

Reply:
we have hard affinity afaik

Reply:
@calvix you are right, after trying a rolling update changing the image the new pod was getting stuck on pending with:
```
Events:
  FirstSeen    LastSeen    Count    From            SubObjectPath    Type        Reason            Message
  ---------    --------    -----    ----            -------------    --------    ------            -------
  3m        15s        15    default-scheduler            Warning        FailedScheduling    No nodes are available that match all of the predicates: PodFitsHostPorts (3), PodToleratesNodeTaints (1).
```
and the update didn't progress. With `maxUnavailable` set to 1 the rolling update works fine, already pushed this change.